lib/matplotlib/backend_bases.py

```diff
@@ -2241,15 +2241,22 @@ def get_default_filetype(cls):
 
     def get_default_filename(self):
         """
-        Return a string, which includes extension, suitable for use as
-        a default filename.
-        """
-        basename = (self.manager.get_window_title() if self.manager is not None
-                    else '')
-        basename = (basename or 'image').replace(' ', '_')
-        filetype = self.get_default_filetype()
-        filename = basename + '.' + filetype
-        return filename
+        Return a suitable default filename, including the extension.
+        """
+        default_basename = (
+            self.manager.get_window_title()
+            if self.manager is not None
+            else ''
+        )
```

Comment:
I think you still need the default `image` case? (might be good to put a test on this as well)
```suggestion
        )
        default_basename = default_basename or 'image'
```

Reply:
It looks like there was a test already, so I'm a bit confused how this passed previously: https://github.com/matplotlib/matplotlib/blob/7275ad95ff615449fccc9c11a51e21fd21e4fb99/lib/matplotlib/tests/test_backend_bases.py#L66